astropy/wcs/src/unit_list_proxy.c

```diff
@@ -201,18 +200,36 @@ PyUnitListProxy_richcmp(
   if (op != Py_EQ && op != Py_NE) {
     Py_RETURN_NOTIMPLEMENTED;
   }
-  lhs = (PyUnitListProxy *)a;
-  rhs = (PyUnitListProxy *)b;
-  int equal = PyObject_RichCompareBool(lhs->unit_class, rhs->unit_class, Py_EQ);
-  if (equal == -1) {
-    return NULL;  // Exception will be set because the rich-compare failed
+
+  {
```

Comment:
That is this `{` tied to? Seems to be orphaned. Am I reading this right?

Reply:
Yep, it's orphaned. It's a so called "block scope" (it creates a new scope for variables). I used it here to separate the code dealing with the pre-condition checks and the actual comparison.

If that's confusing I can remove it again (and put the declarations at the top of the function).

Reply:
If it is not strictly necessary, I would vote to remove it. Or at least state the intention of the block. Thanks!

Reply:
Okay 👍